geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapred/GeoMesaOutputFormat.scala

```diff
@@ -0,0 +1,148 @@
+    AccumuloOutputFormat.setBatchWriterOptions(job, writerConfig)
+}
+
+/**
+ * Output format that turns simple features into mutations and delegates to AccumuloOutputFormat
+ */
+class GeoMesaOutputFormat extends OutputFormat[Text, SimpleFeature] {
+
+  val delegate = new AccumuloOutputFormat
+
+  override def getRecordWriter(ignored: FileSystem, job: JobConf, name: String, progress: Progressable) = {
+  val params = GeoMesaConfigurator.getDataStoreOutParams(job)
+  new GeoMesaRecordWriter(params, delegate.getRecordWriter(ignored, job, name, progress))
+}
+
+  override def checkOutputSpecs(ignored: FileSystem, job: JobConf) = {
+    val params = GeoMesaConfigurator.getDataStoreOutParams(job)
+    if (!AccumuloDataStoreFactory.canProcess(params)) {
+      throw new IOException("Data store connection parameters are not set")
+    }
+    delegate.checkOutputSpecs(ignored, job)
+  }
+}
+
+/**
+ * Record writer for GeoMesa SimpleFeatures.
+ *
+ * Key is ignored. If the feature type for the given feature does not exist yet, it will be created.
+ */
+class GeoMesaRecordWriter(params: Map[String, String], delegate: RecordWriter[Text, Mutation])
+    extends RecordWriter[Text, SimpleFeature] {
+
+  type TableAndMutations = (Text, FeatureToMutations)
+
+  val ds = DataStoreFinder.getDataStore(params).asInstanceOf[AccumuloDataStore]
+
+  val sftCache          = scala.collection.mutable.HashSet.empty[String]
+  val writerCache       = scala.collection.mutable.Map.empty[String, Seq[TableAndMutations]]
+  val encoderCache      = scala.collection.mutable.Map.empty[String, SimpleFeatureEncoder]
+  val indexEncoderCache = scala.collection.mutable.Map.empty[String, IndexValueEncoder]
+
+  override def write(key: Text, value: SimpleFeature) = {
+    val sft = value.getFeatureType
+    val sftName = sft.getTypeName
+
+    // ensure that the type has been created if we haven't seen it before
+    if (sftCache.add(sftName)) {
+      // this is a no-op if schema is already created, and should be thread-safe from different mappers
+      ds.createSchema(value.getFeatureType)
+      // short sleep to ensure that feature is fully written if it is happening in some other thread
+      Thread.sleep(1000)
+    }
+
+    val writers = writerCache.getOrElseUpdate(sftName, {
+      val stEncoder = IndexSchema.buildKeyEncoder(sft, ds.getIndexSchemaFmt(sft.getTypeName))
+      val stWriter = SpatioTemporalTable.spatioTemporalWriter(stEncoder)
+      val stTable = new Text(ds.getSpatioTemporalTable(sft))
+      val recWriter = RecordTable.recordWriter(sft)
+      val recTable = new Text(ds.getRecordTable(sft))
+      AttributeTable.attributeWriter(sft) match {
+        case None => Seq((stTable, stWriter), (recTable, recWriter))
+        case Some(attrWriter) =>
+          val attrTable = new Text(ds.getAttributeTable(sft))
+          Seq((stTable, stWriter), (recTable, recWriter), (attrTable, attrWriter))
```

Comment:
Can we extract this function, shove it in the DataStore or something?

I guess the argument against is that anyone would be able to get access to our writers easily?  

Also, this bypasses any metrics writing.  Hmm....